support-frontend/app/services/MParticleClient.scala

```diff
@@ -0,0 +1,169 @@
+package services
+
+import com.gu.monitoring.SafeLogging
+import com.gu.okhttp.RequestRunners.FutureHttpClient
+import com.gu.rest.WebServiceHelper
+import config.{MparticleConfig, MparticleConfigProvider}
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.syntax._
+import org.joda.time.DateTime
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+case class MParticleUserProfile(
+    hasMobileAppDownloaded: Boolean,
+    hasFeastMobileAppDownloaded: Boolean,
+)
+
+case class MParticleAccessToken(token: String) extends AnyVal
+
+case class OAuthTokenRequest(
+    client_id: String,
+    client_secret: String,
+    audience: String,
+    grant_type: String,
+)
+
+case class OAuthTokenResponse(
+    access_token: String,
```

Comment:
linked 1/3 - the earlier we wrap it, the more type safe it is, also reduces clutter of wrapping/unwrapping
```suggestion
    access_token: MParticleAccessToken,
```